src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+                manufacturer.setName(name);
+                manufacturer.setCountry(country);
+            }
+            return Optional.of(manufacturer);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer by id: "
+                    + id + " from DB.", e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        String getAllManufacturersRequest = "SELECT * FROM manufacturers WHERE is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement =
+                        connection.prepareStatement(getAllManufacturersRequest)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String name = resultSet.getString("name");
+                String country = resultSet.getString("country");
+                Manufacturer manufacturer = new Manufacturer();
+                manufacturer.setId(id);
+                manufacturer.setName(name);
+                manufacturer.setCountry(country);
+                allManufacturers.add(manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers SET name = ?, country = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufactureStatement =
+                        connection.prepareStatement(updateRequest)) {
+            Long id = manufacturer.getId();
+            String name = manufacturer.getName();
+            String country = manufacturer.getCountry();
+            updateManufactureStatement.setString(1, name);
+            updateManufactureStatement.setString(2, country);
+            updateManufactureStatement.setLong(3, id);
+            updateManufactureStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer: "
+                    + manufacturer + " to DB.", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufactureStatement =
+                        connection.prepareStatement(deleteRequest)) {
+            deleteManufactureStatement.setLong(1, id);
+            return deleteManufactureStatement.executeUpdate() >= 1;
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html)
```suggestion
            return deleteManufactureStatement.executeUpdate() > 0;
```